spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
```

Comment:
Shared examples is something that we should avoid IMO. They make a test much harder to read through and debug if something goes wrong. They should only ever be used if some classes contain common behaviour.